rules_gapic/java/java_gapic_pkg.bzl

```diff
@@ -161,182 +155,90 @@ java_gapic_srcs_pkg = rule(
+    client_test_deps = []
+    grpc_deps = []
+    proto_deps = []
+
+    processed_deps = {} #there is no proper Set in Starlark
+    for dep in deps:
+        if _ends_with(dep, "_java_gapic"):
+            _put_dep_in_a_bucket(dep, client_deps, processed_deps)
+            _put_dep_in_a_bucket("%s_test" % dep, client_test_deps, processed_deps)
+            _put_dep_in_a_bucket("%s_resource_name" % dep, proto_deps, processed_deps)
+        elif _ends_with(dep, "_java_grpc"):
+            _put_dep_in_a_bucket(dep, grpc_deps, processed_deps)
+        else:
+            _put_dep_in_a_bucket(dep, proto_deps, processed_deps)
 
     if proto_deps:
-        java_gapic_proto_gradle_pkg(
+        _java_gapic_gradle_pkg(
             name = proto_target,
+            template_label = Label("//rules_gapic/java:resources/gradle/proto.gradle.tmpl"),
             deps = proto_deps,
-            group = grpc_group,
-            version = version,
+            **kwargs
         )
         proto_target_dep = [":%s" % proto_target]
 
     if grpc_deps:
-        java_gapic_grpc_gradle_pkg(
+        _java_gapic_gradle_pkg(
             name = grpc_target,
+            template_label = Label("//rules_gapic/java:resources/gradle/grpc.gradle.tmpl"),
             deps = proto_target_dep + grpc_deps,
-            group = grpc_group,
-            version = version,
+            **kwargs
         )
         grpc_target_dep = ["%s" % grpc_target]
 
-    java_gapic_client_gradle_pkg(
-        name = client_target,
-        deps = proto_target_dep + client_deps,
-        test_deps = grpc_target_dep + client_test_deps,
-        group = client_group,
-        version = version,
-    )
+    if client_deps:
+        _java_gapic_gradle_pkg(
+            name = client_target,
+            template_label = Label("//rules_gapic/java:resources/gradle/client.gradle.tmpl"),
+            deps = proto_target_dep + client_deps,
+            test_deps = grpc_target_dep + client_test_deps,
+            **kwargs
+        )
+        client_target_dep = ["%s" % client_target]
 
-    java_gapic_assembly_gradle_raw_pkg(
+    _java_gapic_assembly_gradle_pkg(
         name = name,
-        deps = proto_target_dep + grpc_target_dep + [":%s" % client_target],
+        deps = proto_target_dep + grpc_target_dep + client_target_dep,
     )
 
-#
-# Private helper functions
-#
-
 def _java_gapic_gradle_pkg(
         name,
-        pkg_type,
+        template_label,
         deps,
-        pkg_deps = [],
-        visibility = None,
         test_deps = None,
-        group = "",
-        version = "",
-        classifier = None):
+        project_deps = None,
+        test_project_deps = None,
+        **kwargs):
     resource_target_name = "%s-resources" % name
-    template_label = Label("//rules_gapic/java:resources/gradle/%s.gradle.tmpl" % pkg_type)
+
+    static_substitutions = dict(_PROPERTIES)
+    static_substitutions["{{name}}"] = name
+    #    static_substitutions.update(extra_substitutions)
```

Comment:
Should we delete this commented code, comment it with a reason or uncomment?

Reply:
I forgot to remove this. Removing now.